File: docs-examples/example-java/src/test/java/io/micronaut/nats/docs/jetstream/kv/KeyValueTest.java
package io.micronaut.nats.docs.jetstream.kv;

import io.micronaut.context.annotation.Property;
import io.micronaut.nats.testcontainers.Nats;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import io.micronaut.test.support.TestPropertyProvider;
import io.nats.client.JetStreamApiException;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInstance;

import java.io.IOException;
import java.util.Map;

@MicronautTest
@Property(name = "spec.name", value = "KeyValueTest")
@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class KeyValueTest implements TestPropertyProvider {

    @Test
    void simpleTest(KeyValueStoreHolder keyValueStoreHolder) throws JetStreamApiException, IOException {
        keyValueStoreHolder.put("test", "myvalue");

        Assertions.assertEquals("myvalue", keyValueStoreHolder.store.get("test").getValueAsString());
    }

    @Override
    public Map<String, String> getProperties() {
        return Nats.getProperties();
    }
}

File: docs-examples/example-python/src/test/python/micronaut/nats/docs/jetstream/kv/KeyValueTest.py
from typing import Annotated

from jakarta.inject import Inject
from micronaut.context.annotation import Property
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

from .KeyValueStoreHolder import KeyValueStoreHolder


@Property(name="spec.name", value="KeyValueTest")
@MicronautTest(environments=["nats"])
class KeyValueTest:
    key_value_store_holder: Annotated[KeyValueStoreHolder, Inject]

    @Test
    def simple_test(self):
        self.key_value_store_holder.put("test", "myvalue")

        assert self.key_value_store_holder.store.get("test").getValueAsString() == "myvalue"
